src/Features/VisualBasic/Portable/ValidateFormatString/VisualBasicValidateFormatStringDiagnosticAnalyzer.vb

```diff
@@ -0,0 +1,130 @@
+﻿'Copyright(c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt In the project root For license information.
+
+Imports Microsoft.CodeAnalysis.Diagnostics
+Imports Microsoft.CodeAnalysis.LanguageServices
+Imports Microsoft.CodeAnalysis.ValidateFormatString
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+Imports System.Collections.Immutable
+Imports System.Linq
+Imports System.Threading
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.ValidateFormatString
+
+    <DiagnosticAnalyzer(LanguageNames.VisualBasic)>
+    Friend Class VisualBasicValidateFormatStringDiagnosticAnalyzer
+        Inherits AbstractValidateFormatStringDiagnosticAnalyzer(Of SyntaxKind)
+
+        Protected Overrides Function GetSyntaxFactsService() As ISyntaxFactsService
+            Return VisualBasicSyntaxFactsService.Instance
+        End Function
+
+        Friend Overrides Function TryGetArgsArgumentType(
+            semanticModel As SemanticModel,
+            arguments As SeparatedSyntaxList(Of SyntaxNode),
+            parameters As ImmutableArray(Of IParameterSymbol),
+            ByRef argsArgumentType As ITypeSymbol) As Boolean
+
+            Dim argsArgument As ArgumentSyntax = Nothing
+            If Not TryGetArgument(semanticModel, "args", arguments, parameters, argsArgument) Then
+                argsArgumentType = Nothing
+                Return False
+            End If
+
+            argsArgumentType = semanticModel.GetTypeInfo(argsArgument.GetArgumentExpression()).Type
+            Return True
+        End Function
+
+        Protected Overrides Function TryGetFormatStringLiteralExpressionSyntax(
```

Comment:
this logic looks the same too.  I'm not sure why we can't just move this up to the base type, using the appropriate syntax facts to work agnostically over c# and vb.